### DISCUSSION TITLE
Failed to map usbserial

### DISCUSSION THREAD

**Discussion by aleblack** [utmapp/UTM#6129] [UTM Help (macOS)]
**Describe the issue**  
Qemu failed to map /dev/tty.usbserial-X inside guest VM

**Configuration**  
* UTM Version: 4.4.5 (94)
* macOS Version: 14.3
* Mac Chip (Intel, M1, ...): M2

I'm trying to map a usb serial port to the VM by adding these lines in the Arguments configuration of utm:
**-chardev serial,path=/dev/tty.usbserial-D201046T,id=digifly -usb -device usb-serial,chardev,digifly**

but I got an error from Qemu on startup:

`QEMU error: QEMU exited from an error: qemu-aarch64-softmmu: -chardev serial,path=/dev/tty.usbserial-D201046T,id=digifly: Could not open '/dev/tty.usbserial-D201046T': Operation not permitted`

but /dev/tty.usbserial-D201046T exist, it's owned by root and has permission of 666:
`❯ ls -la /dev/tty.usbserial-D201046T`
`crw-rw-rw-  1 root  wheel  0x9000002 Feb  8 15:48 /dev/tty.usbserial-D201046T`

What am I doing wrong?

**Comment by DUOLabs333** [comment]
It might be because the user can't access it, and therefore, QEMU needs to run at root. This is hacky, but you could probably fix this by making the QEMU binary in UTM setuid so it automatically runs as root. A better idea may be `chmod 777 <the path>`. Can you open it as a user with `cat`?

**Comment by osy** [comment]
This is not possible as UTM is a sandboxed app. You might be able to do something like create a PTTY device and manually socat that into your USB device. Alternatively, capture the USB device directly in the VM.
